test/functional/eval/api_functions_spec.lua

```diff
@@ -1,11 +1,33 @@
 local helpers = require('test.functional.helpers')(after_each)
 local Screen = require('test.functional.ui.screen')
 local lfs = require('lfs')
+local mpack = require('mpack')
 local neq, eq, execute = helpers.neq, helpers.eq, helpers.execute
 local clear, curbufmeths = helpers.clear, helpers.curbufmeths
 local exc_exec, expect, eval = helpers.exc_exec, helpers.expect, helpers.eval
 local insert = helpers.insert
 
+local read_mpack_file = function(fname)
+  local fd = io.open(fname, 'rb')
+  local data = fd:read('*a')
+  fd:close()
+  local unpack = mpack.Unpacker()
+  return unpack(data)
+end
+
+-- ignore metadata in API function spec
+local remove_function_metadata = function(fspec)
+  fspec['can_fail'] = nil
```

Comment:
`can_fail` `async`, `recieves_channel_id` are now gone from public metadata and don't need to be removed here here. `since` and `method` shouldn't change, so these should not be removed.